### PR TITLE
Fix display of sidetext gates with conditions

### LINKED ISSUE #7532: Classical conditions misalign in both ASCII and MPL output
### Environment

- **Qiskit Terra version**: 0.19.1
- **Python version**:  3.7.12
- **Operating system**: Ubuntu 18.04.5 LTS (Google Colab)

### What is happening?

I was testing my automatic code generation for Qiskit, and apparently the ~cphase~ controlled U1 gate with a classical condition will make the circuit misaligns in both of ASCII output and MPL output.

Screenshots:
```
q2_0: ─■───────■───────■───────■───────■───────■───────■───────■───────■──────»
       │U1(0)  │U1(0)  │U1(0)  │U1(0)  │U1(0)  │U1(0)  │U1(0)  │U1(0)  │U1(0) »
q2_1: ─■───────■───────■───────■───────■───────■───────■───────■───────■──────»
       ║       ║       ║       ║       ║       ║       ║       ║       ║      »
q2_2: ─╫───────╫───────╫───────╫───────╫───────╫───────╫───────╫───────╫──────»
      ┌╫─╨──┐ ┌╫─╨──┐ ┌╫─╨──┐ ┌╫─╨──┐ ┌╫─╨──┐ ┌╫─╨──┐ ┌╫─╨──┐ ┌╫─╨──┐ ┌╫─╨───┐»
c2: 5/╡║0x1 ╞═╡║0x3 ╞═╡║0x5 ╞═╡║0x7 ╞═╡║0x9 ╞═╡║0xb ╞═╡║0xd ╞═╡║0xf ╞═╡║0x11 ╞»
      └╫────┘ └╫────┘ └╫────┘ └╫────┘ └╫────┘ └╫────┘ └╫────┘ └╫────┘ └╫─────┘»
«                                                                         »
«q2_0: ─■───────■───────■───────■───────■───────■───────■───────────■─────»
«       │U1(0)  │U1(0)  │U1(0)  │U1(0)  │U1(0)  │U1(0)  │U1(0) ┌────┴────┐»
«q2_1: ─■───────■───────■───────■───────■───────■───────■──────┤ U2(0,0) ├»
«       ║       ║       ║       ║       ║       ║       ║      └────╥────┘»
«q2_2: ─╫───────╫───────╫───────╫───────╫───────╫───────╫───────────╫─────»
«      ┌╫─╨───┐┌╫─╨───┐┌╫─╨───┐┌╫─╨───┐┌╫─╨───┐┌╫─╨───┐┌╫─╨───┐  ┌──╨──┐  »
«c2: 5/╡║0x13 ╞╡║0x15 ╞╡║0x17 ╞╡║0x19 ╞╡║0x1b ╞╡║0x1d ╞╡║0x1f ╞══╡ 0x2 ╞══»
«      └╫─────┘└╫─────┘└╫─────┘└╫─────┘└╫─────┘└╫─────┘└╫─────┘  └─────┘  »
«                                                                        »
«q2_0: ─────■──────────■──────────■──────────■──────────■──────────■─────»
«      ┌────┴────┐┌────┴────┐┌────┴────┐┌────┴────┐┌────┴────┐┌────┴────┐»
«q2_1: ┤ U2(0,0) ├┤ U2(0,0) ├┤ U2(0,0) ├┤ U2(0,0) ├┤ U2(0,0) ├┤ U2(0,0) ├»
«      └────╥────┘└────╥────┘└────╥────┘└────╥────┘└────╥────┘└────╥────┘»
«q2_2: ─────╫──────────╫──────────╫──────────╫──────────╫──────────╫─────»
«        ┌──╨──┐    ┌──╨──┐    ┌──╨──┐    ┌──╨──┐    ┌──╨──┐    ┌──╨──┐  »
«c2: 5/══╡ 0x3 ╞════╡ 0x6 ╞════╡ 0x7 ╞════╡ 0xa ╞════╡ 0xb ╞════╡ 0xe ╞══»
«        └─────┘    └─────┘    └─────┘    └─────┘    └─────┘    └─────┘  »
«                                                                        »
«q2_0: ─────■──────────■──────────■──────────■──────────■──────────■─────»
«      ┌────┴────┐┌────┴────┐┌────┴────┐┌────┴────┐┌────┴────┐┌────┴────┐»
«q2_1: ┤ U2(0,0) ├┤ U2(0,0) ├┤ U2(0,0) ├┤ U2(0,0) ├┤ U2(0,0) ├┤ U2(0,0) ├»
«      └────╥────┘└────╥────┘└────╥────┘└────╥────┘└────╥────┘└────╥────┘»
«q2_2: ─────╫──────────╫──────────╫──────────╫──────────╫──────────╫─────»
«        ┌──╨──┐    ┌──╨───┐   ┌──╨───┐   ┌──╨───┐   ┌──╨───┐   ┌──╨───┐ »
«c2: 5/══╡ 0xf ╞════╡ 0x12 ╞═══╡ 0x13 ╞═══╡ 0x16 ╞═══╡ 0x17 ╞═══╡ 0x1a ╞═»
«        └─────┘    └──────┘   └──────┘   └──────┘   └──────┘   └──────┘ »
«                                       ┌───────────┐┌───────────┐       »
«q2_0: ─────■──────────■──────────■─────┤ U3(0,0,0) ├┤ U3(0,0,0) ├───X───»
«      ┌────┴────┐┌────┴────┐┌────┴────┐└─────┬─────┘└─────┬─────┘   │   »
«q2_1: ┤ U2(0,0) ├┤ U2(0,0) ├┤ U2(0,0) ├──────■────────────■─────────■───»
«      └────╥────┘└────╥────┘└────╥────┘      ║            ║         │   »
«q2_2: ─────╫──────────╫──────────╫───────────╫────────────╫─────────X───»
«        ┌──╨───┐   ┌──╨───┐   ┌──╨───┐    ┌──╨───┐     ┌──╨───┐  ┌──╨──┐»
«c2: 5/══╡ 0x1b ╞═══╡ 0x1e ╞═══╡ 0x1f ╞════╡ 0x15 ╞═════╡ 0x17 ╞══╡ 0x2 ╞»
«        └──────┘   └──────┘   └──────┘    └──────┘     └──────┘  └─────┘»
«                                                           
«q2_0: ───X──────X──────X──────X───────X───────X───────X────
«         │      │      │      │       │       │       │    
«q2_1: ───■──────■──────■──────■───────■───────■───────■────
«         │      │      │      │       │       │       │    
«q2_2: ───X──────X──────X──────X───────X───────X───────X────
«      ┌──╨──┐┌──╨──┐┌──╨──┐┌──╨───┐┌──╨───┐┌──╨───┐┌──╨───┐
«c2: 5/╡ 0x3 ╞╡ 0xa ╞╡ 0xb ╞╡ 0x12 ╞╡ 0x13 ╞╡ 0x1a ╞╡ 0x1b ╞
«      └─────┘└─────┘└─────┘└──────┘└──────┘└──────┘└──────┘
```

image

### How can we reproduce the issue?

Using the source code below:

```python3
from numpy import pi, e as euler
from qiskit import QuantumCircuit, QuantumRegister, ClassicalRegister
from qiskit.circuit.library.standard_gates import SdgGate, TdgGate, SXGate, RXGate, RYGate, RZGate, U1Gate, U2Gate, U3Gate, SwapGate, XGate, YGate, ZGate, HGate, PhaseGate, SGate, TGate

qr = QuantumRegister(3)
cr = ClassicalRegister(5)
qc = QuantumCircuit(qr, cr)

qc.append(U1Gate(0).control(1), [1, 0]).c_if(cr, 1)
qc.append(U1Gate(0).control(1), [1, 0]).c_if(cr, 3)
qc.append(U1Gate(0).control(1), [1, 0]).c_if(cr, 5)
qc.append(U1Gate(0).control(1), [1, 0]).c_if(cr, 7)
qc.append(U1Gate(0).control(1), [1, 0]).c_if(cr, 9)
qc.append(U1Gate(0).control(1), [1, 0]).c_if(cr, 11)
qc.append(U1Gate(0).control(1), [1, 0]).c_if(cr, 13)
qc.append(U1Gate(0).control(1), [1, 0]).c_if(cr, 15)
qc.append(U1Gate(0).control(1), [1, 0]).c_if(cr, 17)
qc.append(U1Gate(0).control(1), [1, 0]).c_if(cr, 19)
qc.append(U1Gate(0).control(1), [1, 0]).c_if(cr, 21)
qc.append(U1Gate(0).control(1), [1, 0]).c_if(cr, 23)
qc.append(U1Gate(0).control(1), [1, 0]).c_if(cr, 25)
qc.append(U1Gate(0).control(1), [1, 0]).c_if(cr, 27)
qc.append(U1Gate(0).control(1), [1, 0]).c_if(cr, 29)
qc.append(U1Gate(0).control(1), [1, 0]).c_if(cr, 31)
qc.append(U2Gate(0, 0).control(1), [0, 1]).c_if(cr, 2)
qc.append(U2Gate(0, 0).control(1), [0, 1]).c_if(cr, 3)
qc.append(U2Gate(0, 0).control(1), [0, 1]).c_if(cr, 6)
qc.append(U2Gate(0, 0).control(1), [0, 1]).c_if(cr, 7)
qc.append(U2Gate(0, 0).control(1), [0, 1]).c_if(cr, 10)
qc.append(U2Gate(0, 0).control(1), [0, 1]).c_if(cr, 11)
qc.append(U2Gate(0, 0).control(1), [0, 1]).c_if(cr, 14)
qc.append(U2Gate(0, 0).control(1), [0, 1]).c_if(cr, 15)
qc.append(U2Gate(0, 0).control(1), [0, 1]).c_if(cr, 18)
qc.append(U2Gate(0, 0).control(1), [0, 1]).c_if(cr, 19)
qc.append(U2Gate(0, 0).control(1), [0, 1]).c_if(cr, 22)
qc.append(U2Gate(0, 0).control(1), [0, 1]).c_if(cr, 23)
qc.append(U2Gate(0, 0).control(1), [0, 1]).c_if(cr, 26)
qc.append(U2Gate(0, 0).control(1), [0, 1]).c_if(cr, 27)
qc.append(U2Gate(0, 0).control(1), [0, 1]).c_if(cr, 30)
qc.append(U2Gate(0, 0).control(1), [0, 1]).c_if(cr, 31)
qc.append(U3Gate(0, 0, 0).control(1), [1, 0]).c_if(cr, 21)
qc.append(U3Gate(0, 0, 0).control(1), [1, 0]).c_if(cr, 23)
qc.append(SwapGate().control(1), [1, 0, 2]).c_if(cr, 2)
qc.append(SwapGate().control(1), [1, 0, 2]).c_if(cr, 3)
qc.append(SwapGate().control(1), [1, 0, 2]).c_if(cr, 10)
qc.append(SwapGate().control(1), [1, 0, 2]).c_if(cr, 11)
qc.append(SwapGate().control(1), [1, 0, 2]).c_if(cr, 18)
qc.append(SwapGate().control(1), [1, 0, 2]).c_if(cr, 19)
qc.append(SwapGate().control(1), [1, 0, 2]).c_if(cr, 26)
qc.append(SwapGate().control(1), [1, 0, 2]).c_if(cr, 27)

qc.draw() # For ASCII output
# qc.draw('mpl') # For MPL output
```

### What should happen?

They should align in a vertical manner.

### Any suggestions?

_No response_

### PR DESCRIPTION
**Pull request by enavarro51**
### Summary

Fixes #7532

### Details and comments

This PR fixes a problem where the gates with sidetext, such as CPhaseGate and RZZGate, did not line up properly in the text drawer when a condition was set on the gate.

### CODE REVIEW

**Review by javabster — COMMENTED** [review]
This is looking good thanks @enavarro51!

Could you add another test or 2 to check the other side-text type gates with conditionals?

**Comment by enavarro51** [comment]
mpl and latex tests added in https://github.com/Qiskit/qiskit-terra/pull/7673/commits/b706676938b208a97748bf126ac81b5210939712.

**Comment by javabster** [comment]
Sorry Ed I actually meant can you add some more text drawer tests but for _different types_ of sidetext gates. All of these test the `CPhase` gate but would be great to test some of the others (e.g. those mentioned in the other issue) as well

**Comment by enavarro51** [comment]
@javabster I added text drawer tests for cu1 and rzz. I left the mpl and latex tests in even though they work fine, since we didn't have any sidetext with conditions tests for those drawers

**Review by javabster — APPROVED** [review]
Thanks @enavarro51 this LGTM 🚀  snapshot tests all pass in binder

File: test/python/visualization/test_circuit_text_drawer.py [hunk 1]
```diff
@@ -635,6 +637,58 @@ def test_text_cp(self):
...
+                "                      ",
+                "c_2: ═════════════════",
+                "                      ",
+            ]
+        )
+        qr = QuantumRegister(3, "q")
+        cr = ClassicalRegister(3, "c")
+        circuit = QuantumCircuit(qr, cr)
+        circuit.append(gate(pi / 2), [qr[0], qr[1]]).c_if(cr[1], 1)
+        self.assertEqual(str(_text_circuit_drawer(circuit, initial_state=False)), expected)
```
[the remark this answers is not in the document]
Comment (enavarro51) [reply test/python/visualization/test_circuit_text_drawer.py]: Done in https://github.com/Qiskit/qiskit-terra/pull/7673/commits/d0cfe734ebed6084e8d3d4ceda72635de700eab2

**Review by 1ucian0 — APPROVED** [review]
Thanks! Great work!

**Pull request merged**